### PR TITLE
Add Type Annotations for Improved Type Safety in Windows

### PR DESCRIPTION
**Pull request by ashmeigh**
### Description

Added type annotations to various functions and methods within the mantidimaging/gui/windows directory to enhance type safety and improve code maintainability. These annotations ensure clearer function signatures and better compatibility with Python type checkers like mypy.

### Testing 

Type annotations do not introduce runtime errors.
Functionality across affected modules remains intact.
All affected code paths are covered by existing unit tests.

### Acceptance Criteria 

Mannual testing 
No linting errors related to type annotations.
No type-related runtime errors occur during testing.
All affected code paths are adequately covered by existing tests.

### CODE REVIEW

File: mantidimaging/gui/windows/welcome_screen/presenter.py [hunk 1]
```diff
@@ -55,26 +55,26 @@ def show_today():
         else:
             return show_at_start
 
-    def set_up_links(self):
+    def set_up_links(self) -> None:
         for link_name, url in WELCOME_LINKS:
             self.add_link(link_name, url)
 
-    def add_link(self, link_name, url):
+    def add_link(self, link_name, url) -> None:
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/welcome_screen/presenter.py:62]: Argument types, probably `str`

File: mantidimaging/gui/windows/welcome_screen/view.py [hunk 2]
```diff
@@ -25,21 +25,21 @@ def __init__(self, parent, presenter):
...
-    def get_show_at_start(self):
+    def get_show_at_start(self) -> None:
         return self.show_at_start.isChecked()
 
-    def set_show_at_start(self, checked):
+    def set_show_at_start(self, checked) -> None:
         self.show_at_start.setChecked(checked)
 
-    def set_version_label(self, contents):
+    def set_version_label(self, contents) -> None:
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/welcome_screen/view.py:34]: Argument types

File: mantidimaging/gui/windows/welcome_screen/view.py [hunk 3]
```diff
@@ -25,21 +25,21 @@ def __init__(self, parent, presenter):
...
         self.version_label.setText(contents)
 
-    def add_link(self, label, row):
+    def add_link(self, label, row) -> None:
         link_label = QLabel(label)
         link_label.setOpenExternalLinks(True)
         self.link_box_layout.addWidget(link_label, row, 0)
 
-    def add_issues(self, contents):
+    def add_issues(self, contents) -> None:
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/welcome_screen/view.py:42]: And here

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
There are a few argument types that could be added aswell.

**Review by MikeSullivan7 — APPROVED** [review]
Looks good, all tests pass and MI runs as normal with no errors.

**Review by samtygier-stfc — APPROVED** [review]

**Pull request merged**